src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/AccountSupervisor.kt

```diff
@@ -593,25 +594,24 @@ internal open class AccountSupervisor(
     }
 
     private fun handleComplianceResponse(response: String?, httpCode: Int): ComplianceStatus {
-        compliance = if (helper.success(httpCode) && response != null) {
+        var complianceStatus = ComplianceStatus.UNKNOWN
+        var updatedAt: String? = null
+        var expiresAt: String? = null
+        if (helper.success(httpCode) && response != null) {
             val res = helper.parser.decodeJsonObject(response)?.toIMap()
-            if (res != null) {
-                val status = helper.parser.asString(res["status"])
-                val updatedAt = helper.parser.asString(res["updatedAt"])
-                val complianceStatus =
-                    if (status != null) {
-                        ComplianceStatus.valueOf(status)
-                    } else {
-                        ComplianceStatus.UNKNOWN
-                    }
-                Compliance(compliance.geo, complianceStatus, updatedAt ?: compliance.updatedAt)
-            } else {
-                Compliance(compliance.geo, ComplianceStatus.UNKNOWN, compliance.updatedAt)
+            complianceStatus = helper.parser.asString(res?.get("status"))?.let { ComplianceStatus.valueOf(it) } ?: ComplianceStatus.UNKNOWN
+            updatedAt = helper.parser.asString(res?.get("updatedAt"))
+            if (updatedAt != null) {
+                expiresAt = try {
+                    Instant.parse(updatedAt).plus(7.days).toString()
+                } catch (e: IllegalArgumentException) {
+                    Logger.e { "Error parsing compliance updatedAt: $updatedAt" }
+                    null
+                }
             }
-        } else {
-            Compliance(compliance.geo, ComplianceStatus.UNKNOWN, compliance.updatedAt)
         }
-        return compliance.status
+        compliance = Compliance(compliance.geo, complianceStatus, updatedAt, expiresAt)
+        return complianceStatus
```

Comment:
@rosepuppy  this should be same functionality, just cleaned up a little so that we have less verboseness in creating the Compliance object. Same thing in the abacus v1 implementation

please ensure u see no different conditional paths

Reply:
Awesome, thanks for cleaning this up!!